Learn - IOTA Messages Updates (#513)
iota-community/iota-wiki: internal/learn/about-iota/messages.md

---
id: messages
title: Messages
description: This topic explores messages, models, and payloads that encompass the transfer of data within the Tangle.
---

# Messages

A message is an object that nodes gossip around in the network. It always references one to eight other messages, which are known as **parents**. It is stored as a vertex on the Tangle data structure maintained by the nodes.

Messages can contain payloads. Some of them are core payloads that are processed by all nodes as part of the core protocol. Others are community payloads that build up new functionalities on top of the Tangle. And some payloads have other nested payloads embedded inside. So, the parsing of the message is done layer by layer, to ensure embedded payloads also have a correct syntax structure.

## UTXO

> Previously, the IOTA protocol used transactions (which were vertices in the Tangle), where each transaction defined either an input or an output. A grouping of those input and output transaction vertices made up a bundle that transferred the given values as an atomic unit. But this approach was seen as too time-consuming. So, we adopted a new transaction structure called the **unspent transaction outputs **(UTXO)**.

Reviewer: ```suggestion
Previously, the IOTA protocol used transactions (which were vertices in the Tangle), where each transaction defined either an input or an output. A grouping of those input and output transaction vertices made up a bundle that transferred the given values as an atomic unit. But this approach was seen as too time-consuming. So, we adopted a new transaction structure called the **unspent transaction outputs (UTXO)**.
```

The UTXO model defines a ledger state where balances are not directly associated with addresses but with the outputs of transactions. In this model, transactions specify the outputs of previous transactions as inputs, which are consumed to create new outputs. A transaction must consume the entirety of the specified inputs.

![utxo-model](/img/learn/about-iota/utxo.png)

> So, the UTXO is a part of a larger, self-contained, and flexible message structure known as a **payload**. This approach is helps enable a self-contained message structure defining the data of the entire transfer as a payload to be embedded into a message.

Reviewer: ```suggestion
So, the UTXO is a part of a larger, self-contained, and flexible message structure known as a **payload**. This approach helps to enable a self-contained message structure defining the data of the entire transfer as a payload to be embedded into a message.
```

## Message Payloads

### Transaction Payload

A transaction payload (defined in [TIP-0020](https://github.com/lzpap/tips/blob/tx-updates/tips/TIP-0020/tip-0020.md)) is made up of two parts:

1. The Transaction Essence part, which contains the inputs, outputs, and an optional embedded payload.
2. The Unlock Blocks, which unlocks the Transaction Essence's inputs. In case the unlock block contains a signature, it signs the entire Transaction Essence part.

> In general, all parts of a transaction payload begin with a byte describing the type of a given part. This helps keep the flexibility of introducing new types or versions of the given part in the future.

Reviewer: ```suggestion
In general, all parts of a transaction payload begin with a byte describing the type of a given part. This allows the introduction of new types or versions of the given part in the future.
```
